src/Aspire.Dashboard/Components/Pages/Resources.razor.cs

```diff
@@ -97,17 +96,29 @@ protected override void OnInitialized()
             await InvokeAsync(StateHasChanged);
         });
 
-        void SubscribeResources()
+        async Task SubscribeResources()
         {
             var (snapshot, subscription) = DashboardClient.SubscribeResources();
 
             // Apply snapshot.
             foreach (var resource in snapshot)
             {
                 var added = _resourceByName.TryAdd(resource.Name, resource);
+
+                ImmutableInterlocked.Update(ref _allResourceTypes, set => set.Add(resource.ResourceType));
                 Debug.Assert(added, "Should not receive duplicate resources in initial snapshot data.");
             }
 
+            await InvokeAsync(() =>
```

Comment:
Is this necessary? This is called directly from OnInitializedAsync so it should already be on the UI thread.

Reply:
Works, and I can make this sync again.